token/src/main/java/edu/tamu/weaver/token/service/TokenService.java

```diff
@@ -130,4 +151,17 @@ public String craftToken(Map<String, String> headers) throws InvalidKeyException
         return createToken(subject, claims);
     }
 
+    private String sha512Secret(String input) throws NoSuchAlgorithmException {
+        MessageDigest md = MessageDigest.getInstance(MessageDigestAlgorithms.SHA_512);
+        byte[] digest = md.digest(input.getBytes());
+        BigInteger no = new BigInteger(1, digest);
+        String hash = no.toString(16);
+
+        while (hash.length() < 32) {
+            hash = "0" + hash;
+        }
```

Comment:
I'm not following this logic.

If length < 32, why prepend a 0.

Reply:
It ensures it is a 512 bit to use with HS512 signature algorithm.

Reply:
So the confusing part in this service is that the secret was used for both signing the JWT and then encrypting the JWT. With the jsonwebtoken upgrade the methods we used have been deprecated and require a key or key pair now to sign the JWT. I just added this hack to hash the secret to a 512 bit secret.